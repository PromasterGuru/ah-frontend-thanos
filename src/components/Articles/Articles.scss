.articles-body {
  display: flex;
  flex-flow: row wrap;

  padding: 30px;

  .article-card {
    min-width: 250px;
    max-width: 25%;
    padding: 20px;
    justify-content: center;
    display: flex;
    flex-wrap: wrap;
  }

  .image-wrapper {
    max-width: 100%;
    max-height: 200px;
    overflow: hidden;
    margin: 0;
    border-radius: 20px;

    img {
      display: block;
      margin: 0 -30%;
      width: 177.777%;
    }
  }
  .text-section {
    max-width: 90%;
    margin-top: -50px;
    z-index: 2;
    background-color: white;
    border-radius: 30px;
    padding: 20px;
    box-shadow: 2px 10px #efefef;

    .title-section {
    }
    .body-section {
    }
    .date-section {
    }
    .read-more {
    }
  }
}
