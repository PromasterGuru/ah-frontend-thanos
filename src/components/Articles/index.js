import React from 'react';
import PropTypes from 'prop-types';

const ArticleDisplay = ({ article }) => (
  <div className="article-card">
    <div className="image-wrapper">
      <img src={article.image_url} alt="poster" />
    </div>
    <div className="text-section">
      <div className="title-section">{article.title}</div>
      <div className="body-section">{article.body.slice(0, 150)}</div>
      <div className="date-section">{article.created_at}</div>
      <button className="read-more" type="button">
        Read More
      </button>
    </div>
  </div>
);
ArticleDisplay.propTypes = {
  article: PropTypes.shape({}).isRequired,
};

export default ArticleDisplay;
