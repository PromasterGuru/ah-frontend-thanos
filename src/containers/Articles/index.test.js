import React from 'react';
import { shallow, mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import Articles from './index';
import ArticleDisplay from '../../components/Articles';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('<Articles />', () => {
  let articlesComponent;

  beforeEach(() => {
    const store = mockStore({ data: { articleReducer: { articles: [] } } });
    articlesComponent = mount(
      <Provider store={store}>
        <Articles data={jest.fn} dispatch={store.dispatch} match={jest.fn} />
      </Provider>
    );
  });

  it('should render the Articles component', () => {
    expect(articlesComponent).toMatchSnapshot();
  });
});

let store;
describe('', () => {
  store = mockStore({});
  it('should render the ArticleDisplay component', () => {
    mount(
      <Provider store={store}>
        <ArticleDisplay
          article={
            ({ body: { slice: jest.fn } }, { image_url: '' }, { title: '' })
          }
        />
      </Provider>
    );
  });
});
