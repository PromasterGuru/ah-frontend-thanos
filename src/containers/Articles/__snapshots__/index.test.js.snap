// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Articles /> should render the Articles component 1`] = `undefined`;
